### ISSUE TITLE
LSP server crashes when using custom json schema with vscode extension v0.19.0 (taplo/lsp v0.6.0)

### ISSUE THREAD

**Issue by ed-curran** [tamasfe/taplo#355]
Hey, thanks for your work on this extension / toolset particularly the LSP.

I've been trying to get the LSP to work with monaco in a browser and got some panics inside the LSP. At first I thought it was me because I wasn't getting them in vscode, however I then realized I was using v0.14.2 of the extension which uses an earlier version of the LSP and toolset etc. So then I installed v0.19.0 of the vscode extension and found I could reproduce the LSP crashes inside vscode. So must be something more fundamental. 

The LSP crash manifests in vscode as vscode trying to restart the LSP, but getting stuck in a crash loop and then eventually it shows a little pop up.
```
The Even Better TOML LSP server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.
```
But the real thing I'm interested in is why the LSP crashed in the first instance, although it raises the side question of why the LSP server fails to restart .

 I can get the LSP to crash rather easily when using a custom json schema. Here's a test schema:
 
 ```json
{
    "$ref": "#/definitions/TestSchema",
    "definitions": {
      "TestSchema": {
        "type": "object",
        "required": [
          "statements"
        ],
        "properties": {
          "statements": {
            "title": "stataments",
            "description": "statements description",
            "type": "array",
            "items": [
              {
                "title": "Create",
                "type": "object",
                "properties": {
                  "create": {
                    "description": "description for create object",
                    "title": "create",
                    "type": "object",
                    "properties": {
                      "a": {
                        "description": "description for property a",
                        "type": "string"
                      }
                    },
                    "required": [
                      "a"
                    ],
                    "additionalProperties": false
                  }
                },
                "required": [
                  "create"
                ],
                "additionalProperties": false
              }
            ]
          }
        }
      }
    }
  }
  
```

I can get the lsp to crash under these conditions:

1. Request an auto completion (e.g. ctrl space shortcut) at the end of line 2
```toml
#:schema ./testSchema.json
[]
```
gets this error in extension logs
```
panicked at 'called `Option::unwrap()` on a `None` value', /home/runner/work/taplo/taplo/crates/taplo/src/util/mod.rs:169:10

Stack:

Error
    at AA.w.wbg.__wbg_new_693216e109162396 (/home/***/.vscode/extensions/tamasfe.even-better-toml-0.19.0/dist/server.js:2:5792)
    at wasm://wasm/00f689f6:wasm-function[5286]:0x320b3a
    at wasm://wasm/00f689f6:wasm-function[2552]:0x2dd09e
    at wasm://wasm/00f689f6:wasm-function[4322]:0x3163e4
    at wasm://wasm/00f689f6:wasm-function[4122]:0x312ad9
    at wasm://wasm/00f689f6:wasm-function[501]:0x1c9b5c
    at wasm://wasm/00f689f6:wasm-function[61]:0x103f3
    at wasm://wasm/00f689f6:wasm-function[102]:0x9c0a7
    at wasm://wasm/00f689f6:wasm-function[3144]:0x2f9465
    at wasm://wasm/00f689f6:wasm-function[74]:0x518ee
```

2. Request an auto completion at the start of line 4
```toml
#:schema ./testSchema.json
[[statements]]
[statements.create]

```
gets error
```
panicked at 'called `Option::unwrap()` on a `None` value', /home/runner/work/taplo/taplo/crates/taplo-lsp/src/query.rs:259:37

Stack:

Error
    at AA.w.wbg.__wbg_new_693216e109162396 (/home/***/.vscode/extensions/tamasfe.even-better-toml-0.19.0/dist/server.js:2:5792)
    at wasm://wasm/00f689f6:wasm-function[5286]:0x320b3a
    at wasm://wasm/00f689f6:wasm-function[2552]:0x2dd09e
    at wasm://wasm/00f689f6:wasm-function[4322]:0x3163e4
    at wasm://wasm/00f689f6:wasm-function[4122]:0x312ad9
    at wasm://wasm/00f689f6:wasm-function[406]:0x1a3b59
    at wasm://wasm/00f689f6:wasm-function[61]:0x145f3
    at wasm://wasm/00f689f6:wasm-function[102]:0x9c0a7
    at wasm://wasm/00f689f6:wasm-function[3144]:0x2f9465
    at wasm://wasm/00f689f6:wasm-function[74]:0x518ee
```

I also noticed that in v0.14.2 I would have descriptions from the json schema show on hover, but this doesn't happen in v0.19.0. 

My feeling is that the position of the cursor is getting out of sync with the state of the DOM, but that's just a guess. In my monaco troubleshooting I tried calling the `textDocument/semanticTokens/full` method on the LSP to get a look into the parsed representation, but it always returns an empty array regardless of the text of the editor which surprised me. I don't know if this is significant or not, I'm not familiar with LSP so I don't know what would be expected.

**Comment by xavdid** [comment]
I'm seeing this as well while setting files in a `starship.toml` file. They [recommend](https://starship.rs/config/) starting the file with:

```toml
# Get editor completions based on the config schema
"$schema" = 'https://starship.rs/config-schema.json'
```

**Comment by chardon55** [comment]
0.19.0 works fine for my Codium, but 0.19.1 keeps crashing after selecting a schema

**Comment by ia0** [comment]
This looks like a duplicate of #434 which has been fixed by #462 and waiting #473 to get released.
